ui/src/app/base/hooks/base.ts

```diff
@@ -87,32 +87,38 @@ export const useCycled = (
  * @param onComplete - The function to call when all the items have been processed.
  * @param hasErrors - Whether there are any item errors in state.
  * @param onError - The function to call when an error occurs.
+ * @returns Whether the list of items has been processed successfully.
  */
-export const useProcessing = (
-  processingCount: number,
-  onComplete: () => void,
+export const useProcessing = ({
   hasErrors = false,
-  onError: () => void
-): void => {
-  const processingStarted = useRef(false);
-  if (processingStarted.current === false && processingCount > 0) {
-    processingStarted.current = true;
-  }
+  onComplete = () => null,
+  onError = () => null,
```

Comment:
I think maybe we shouldn't set default functions here. It might be easier to understand that these are optional if we check for the existence of the function when we call them below.

Reply:
Fair enough! Updated